### ISSUE TITLE
Pruning with artifact caching

### ISSUE THREAD

**Issue by balopat** [GoogleContainerTools/skaffold#3893]
Currently artifact caching and pruning are either-or. Artifact caching is turned on by default resulting in cool speedups but pruning is turned off and Docker VM disks are filling up fasts.

I propose a "limited memory" artifact caching solution that "remembers" only 2-3 images for a given artifact. Also we can add a TTL for our cache too. 

This way we can prune all the skaffold built layers that are - in the artifact cache but out of TTL or beyond the cache capacity for a given artifact. 

Size of the cache and TTL can be configurable.

**Comment by dgageot** [comment]
This whole pruning needs to be re-designed. We should start by pruning only the images we produce

**Comment by dgageot** [comment]
I think it would make sense to run something like `docker image prune --filter "label=built-by=skaffold" --filter "until=30m" --force` but then, we need to make sure all the image we built have this label applied.

**Comment by nkubala** [comment]
yeah I really hate the way I designed pruning...not my finest. I wouldn't mind labelling all of the images we build, we track them anyway so I don't think it would be too much work. what about intermediate images for multi-stage builds though? I think there's a bug where sometimes these get left behind but I'm not sure
